tests/test_format.py

```diff
@@ -70,6 +72,56 @@ async def test_format_code_with_syntax_error(tmp_path, ruff_version: Version):
     )
 
     with handle_unsupported:
-        result = await _run_format_on_document(document, settings)
+        result = await _run_format_on_document(document, settings, None)
         assert result is not None
         assert result.exit_code == 2
+
+
+@pytest.mark.asyncio
+async def test_format_range(tmp_path, ruff_version: Version):
+    original = """x   = 1
+
+
+
+print( "Formatted")
+
+print ("Not formatted")
+"""
+
+    expected = """x   = 1
+
+
+print("Formatted")
+
+print ("Not formatted")
+"""
+
+    test_file = tmp_path.joinpath("main.py")
+    test_file.write_text(original)
+    uri = utils.as_uri(str(test_file))
+
+    workspace = Workspace(str(tmp_path))
+    document = Document.from_text_document(workspace.get_text_document(uri))
+    settings = _get_settings_by_document(document.path)
+
+    handle_unsupported = (
+        pytest.raises(RuntimeError, match=f"Ruff .* required, but found {ruff_version}")
+        if not VERSION_REQUIREMENT_RANGE_FORMATTING.contains(ruff_version)
+        else nullcontext()
+    )
+
+    with handle_unsupported:
```

Comment:
It just comes down to stylistic preference, really, but you could consider using an [`ExitStack`](https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack) here instead of the `nullcontext()` ternary:

```suggestion
    with ExitStack() as stack;
        if not VERSION_REQUIREMENT_RANGE_FORMATTING.contains(ruff_version):
            stack.enter_context(
                pytest.raises(
                    RuntimeError, match=f"Ruff .* required, but found {ruff_version}"
                )
            )

```

Reply:
Uhm, I just copied what the other tests did. I'll leave it for now for consistency

Reply:
Sure, makes sense!